lib/staging.js

```diff
@@ -0,0 +1,457 @@
+import fs from 'node:fs';
+import semver from 'semver';
+import path from 'node:path';
+import { getNcuDir } from './config.js';
+import { readFile, readJson, writeFile, writeJson } from './file.js';
+import Session from './session.js';
+import { forceRunAsync, runSync } from './run.js';
+const isWindows = process.platform === 'win32';
+
+const validNumber = (n) => !Number.isNaN(n) && n > -1 && n < Infinity;
```

Comment:
```suggestion
const validateNumber = (n) => !Number.isNaN(n) && n > -1 && n < Infinity;
```

Just to make it equal to Node.js codebase :)